pinot-core/src/main/java/org/apache/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java

```diff
@@ -50,22 +51,30 @@
   private static SSLContext _sslContext;
   private static Integer _controllerHttpsPort;
 
+  private static boolean _commitWithMetadata;
+
   private final FileUploadDownloadClient _fileUploadDownloadClient;
   private final ServerMetrics _serverMetrics;
 
-  public static void init(Configuration uploaderConfig) {
+  public static void init(Configuration uploaderConfig, boolean commitWithMetadata) {
```

Comment:
Can you move the config of commitWithMetadata to LLRealtimeSegmentDataManager? It is being checked in the other class, so makes sense to move it there. You may have a bit more config plumbing to do, but it is worth it

Reply:
Done. Put the config into IndexLoadingConfig just like splitCommit.